File: src/weaponsprocurement/gui/WimGuiModalActionRow.java
package weaponsprocurement.gui;

import com.fs.starfarer.api.ui.CustomPanelAPI;

import java.util.List;

final class WimGuiModalActionRow {
    private WimGuiModalActionRow() {
    }

    static <A> void add(CustomPanelAPI parent,
                        WimGuiModalLayout modal,
                        float headerHeight,
                        float headerToActionGap,
                        float buttonHeight,
                        float buttonGap,
                        List<WimGuiButtonSpec<A>> specs,
                        List<WimGuiButtonBinding<A>> bindings) {
        float y = modal.actionRowY(headerHeight, headerToActionGap);
        WimGuiControls.addButtonRow(parent, modal.padding, y, buttonHeight, buttonGap, specs, bindings);
    }
}

File: src/main/kotlin/weaponsprocurement/gui/WimGuiModalActionRow.kt
package weaponsprocurement.gui

import com.fs.starfarer.api.ui.CustomPanelAPI

class WimGuiModalActionRow private constructor() {
    companion object {
        @JvmStatic
        fun <A> add(
            parent: CustomPanelAPI,
            modal: WimGuiModalLayout,
            headerHeight: Float,
            headerToActionGap: Float,
            buttonHeight: Float,
            buttonGap: Float,
            specs: List<WimGuiButtonSpec<A>>,
            bindings: List<WimGuiButtonBinding<A>>,
        ) {
            val y = modal.actionRowY(headerHeight, headerToActionGap)
            WimGuiControls.addButtonRow(parent, modal.padding, y, buttonHeight, buttonGap, specs, bindings)
        }
    }
}
